Plugins/NativeEngine/Source/ShaderCompilerTraversers.cpp

```diff
@@ -446,13 +449,49 @@ namespace Babylon::ShaderCompilerTraversers
                 }
             }
 
+#if __APPLE__ | APIOpenGL
+            // This table is a copy of the table bgfx uses for vertex attribute -> shader symbol association.
+            // copied from renderer_gl.cpp.
+            constexpr static const char * s_attribName[] =
+            {
+                "a_position",
+                "a_normal",
+                "a_tangent",
+                "a_bitangent",
+                "a_color0",
+                "a_color1",
+                "a_color2",
+                "a_color3",
+                "a_indices",
+                "a_weight",
+                "a_texcoord0",
+                "a_texcoord1",
+                "a_texcoord2",
+                "a_texcoord3",
+                "a_texcoord4",
+                "a_texcoord5",
+                "a_texcoord6",
+                "a_texcoord7",
+            };
+            BX_STATIC_ASSERT(bgfx::Attrib::Count == BX_COUNTOF(s_attribName) );
+#endif
+
             std::pair<unsigned int, const char*> GetVaryingLocationAndNewNameForName(const char* name)
             {
 #define IF_NAME_RETURN_ATTRIB(varyingName, attrib, newName)  \
     if (std::strcmp(name, varyingName) == 0)                 \
     {                                                        \
         return {static_cast<unsigned int>(attrib), newName}; \
     }
+#if __APPLE__ | APIOpenGL
+                // For OpenGL and Metal platforms, we have an issue where we have a hard limit on the number shader attributes supported.
+                // To work around this issue, instead of mapping our attributes to the most similar bgfx::attribute, instead replace
+                // the first attribute encountered with the symbol bgfx uses for attribute 0 and increment for each subsequent attribute encountered.
+                // This will cause our shader to have nonsensical naming, but will allow us to efficiently "pack" the attributes.
+                UNUSED(name);
+                m_genericAttributesRunningCount++;
+                return {static_cast<unsigned int>(m_genericAttributesRunningCount-1), s_attribName[std::min(static_cast<unsigned int>(m_genericAttributesRunningCount-1), static_cast<unsigned int>(bgfx::Attrib::Count)-1)]};
```

Comment:
if m_genericAttributesRunningCount is bigger than bgfx::Attrib::Count, we should probably throw or something. Silently using the last index doesn't seem right.

Reply:
Ok, that seems more responsible than reusing the last index for multiple attributes.

Reply:
If you don't mind, also fix the UV one.

Reply:
I changed the other attribute location choosing code path to do this check as well.